include/SH3/system/input_config.hpp

```diff
@@ -0,0 +1,31 @@
+/** @file
+ *  Defines configuration variables for the @ref sh3::system::input_system.
+ */
+#ifndef SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+#define SH3_SYSTEM_INPUT_CONFIG_HPP_INCLUDED
+
+/**
+ *  Whether the @ref sh3::system::input_system should provide timing support for inputs.
+ *  
+ *  With this enabled, the @ref sh3::system::input_system keeps track what fraction of a frame the actions were active for.
+ *  
+ * @note Doesn't work right with SDL timestamps.
+ *       I think the event times are not when the input *happened*, but when the event-loop *received* them, which is not useful to us.
+ */
+#ifdef DOXYGEN
+#define INPUT_PROVIDE_TIMING 0
```

Comment:
So the timestamp for each keypress is incorrect due to `SDL_PollEvent()` etc? What functionality are we sacrificing by not having this? IIRC, Silent Hill doesn't really depend on any timed input (thank God).

Also, it would seem you are correct. The timestamp occurs when the `SDL_Event` is actually handled (which is kind of annoying)

https://www.gamedev.net/forums/topic/682928-how-good-are-sdl-event-timestamps/